dessia_common/report.py

```diff
@@ -0,0 +1,199 @@
+            file = open(self.name_report + '.log', 'r')
+        elif option == 'w':
+            file = open(self.name_report + '.log', 'w')
+        return file
+
+    def close(self, file):
+        file.close()
+
+    def add_lines(self, lines: List[str], offset: int = 0, nb_line_blank: int = 0):
+        for line in range(nb_line_blank):
+            lines.append('')
+        for line in lines:
+            self.content += offset*'' + line + '  \n  '
+
+            print(line)
+
+        # self.close(file)
+        # TODO : si besoin de l'export txt, j'ajoute que les lignes nécessaire
+        # with open .... f_stream, f_stream.write en allant à la derniere ligne
+        self.to_txt()
+
+    def add_title(self, title: str):
+        self.last_offset = 0
+        title = title.upper()
+        lines = ['# ' + title]
+
+        self.add_lines(lines, nb_line_blank=2)
+
+    def add_subtitle(self, title: str):
+        self.last_offset = 0
+        title = title[0].upper() + title[1:]
+        if len(title) > self.width_line:
+            title = title[0: self.width_line]
+
+        lines = ['', '## ' + title]
+        self.add_lines(lines, nb_line_blank=1)
+
+    def add_subsubtitle(self, title: str):
+        offset = 2
+        self.last_offset = offset
+        title = title[0].upper() + title[1:]
+        if len(title) > self.width_line - offset:
+            title = title[0: self.width_line - offset]
+
+        lines = [' ' * offset + '### ' + title]
+        self.add_lines(lines, offset, nb_line_blank=0)
+
+    def add_text(self, text: str):
+        offset = self.last_offset + 2
+        line_length = self.width_line - offset
+
+        nb_bloc = text.split(' ')
+        lines = []
+        line = ' ' * offset
+        for bloc in nb_bloc:
+            if bloc == "\n":
+                lines.append(line)
+                line = ' ' * offset
+
+            elif len(line) + len(bloc) <= line_length:
+                line += bloc
+                if len(line) + 1 <= line_length:
+                    line += ' '
+            else:
+                if len(line) == offset:
+                    line += bloc
+                    lines.append(line)
+                    line = ' ' * offset
+                else:
+                    lines.append(line)
+                    line = ' ' * offset + bloc + ' '
+
+        lines.append(line)
+        self.add_lines(lines, offset)
+
+    def add_table(self, title: List[str], elements: List[List[any]]):
+
+        # | Left columns  | Right columns |
+        # | ------------- |:-------------:|
+        # | left foo      | right foo     |
+        # | left bar      | right bar     |
+        # | left baz      | right baz     |
+
+        offset = self.last_offset + 2
+        # line_length = self.width_line - offset
+
+        real_line_length = self.width_line - offset - (len(title) + 1) - 2 * len(title)
+        if real_line_length < 0:
+            raise KeyError('too many columns in the report')
```

Comment:
Thanks for the advice, I wanted to avoid the management when the table would be bigger than the size of the line, I will think about that in a future (near or not)